src/external_trait_impls/rayon/raw.rs

```diff
@@ -0,0 +1,539 @@
+        ParValuesMut { table: self }
+    }
+}
+
+unsafe impl<'a, K, V: Send> Send for ParValuesMut<'a, K, V> {}
+
+impl<'a, K, V: Send> ParallelIterator for ParValuesMut<'a, K, V> {
+    type Item = &'a mut V;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+    where
+        C: UnindexedConsumer<Self::Item>,
+    {
+        let buckets = unsafe { self.table.iter().iter };
+        let producer = ParValuesMutProducer::from(buckets);
+        plumbing::bridge_unindexed(producer, consumer)
+    }
+}
+
+impl<'a, K, V: fmt::Debug> fmt::Debug for ParValuesMut<'a, K, V> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let iter = unsafe { self.table.iter() };
+        let entries = iter.map(|bucket| unsafe {
+            &bucket.as_ref().1
+        });
+
+        f.debug_list().entries(entries).finish()
+    }
+}
+
+struct ParValuesMutProducer<'a, K: 'a, V: 'a> {
+    iter: RawIterRange<(K, V)>,
+    // To ensure invariance with respect to V
+    marker: PhantomData<&'a mut V>,
+}
+
+impl<'a, K, V> From<RawIterRange<(K, V)>> for ParValuesMutProducer<'a, K, V> {
+    #[inline]
+    fn from(iter: RawIterRange<(K, V)>) -> Self {
+        Self { iter, marker: PhantomData }
+    }
+}
+
+unsafe impl<'a, K, V: Send> Send for ParValuesMutProducer<'a, K, V> {}
+
+impl<'a, K, V: Send> UnindexedProducer for ParValuesMutProducer<'a, K, V> {
+    type Item = &'a mut V;
+
+    fn split(self) -> (Self, Option<Self>) {
+        unsafe { raw_iter_range_split_and_convert(self.iter) }
+    }
+
+    fn fold_with<F>(self, folder: F) -> F
+    where
+        F: Folder<Self::Item>,
+    {
+        let iter = self.iter.map(|bucket| unsafe {
+            &mut bucket.as_mut().1
+        });
+        folder.consume_iter(iter)
+    }
+}
+
+
+/// Parallel iterator over entries of a consumed map.
+///
+/// This iterator is created by the [`into_par_iter`] method on [`HashMap`]
+/// (provided by the [`IntoParallelIterator`] trait).
+/// See its documentation for more.
+///
+/// [`into_par_iter`]: /hashbrown/struct.HashMap.html#method.into_par_iter
+/// [`HashMap`]: /hashbrown/struct.HashMap.html
+/// [`IntoParallelIterator`]: https://docs.rs/rayon/1.0/rayon/iter/trait.IntoParallelIterator.html
+pub struct IntoParIter<K, V> {
+    table: RawTable<(K, V)>,
+}
+
+impl<K: Send, V: Send> IntoParallelIterator for RawTable<(K, V)> {
+    type Item = (K, V);
+    type Iter = IntoParIter<K, V>;
+
+    #[inline]
+    fn into_par_iter(self) -> Self::Iter {
+        IntoParIter { table: self }
+    }
+}
+
+impl<K: Send, V: Send> ParallelIterator for IntoParIter<K, V> {
+    type Item = (K, V);
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+    where
+        C: UnindexedConsumer<Self::Item>,
+    {
+        let iter = unsafe { self.table.iter().iter };
+        let alloc = self.table.into_alloc();
+        let producer = IntoParIterProducer { iter, alloc };
+        plumbing::bridge_unindexed(producer, consumer)
```

Comment:
I would arrange for `dealloc` after the bridge returns here, rather than doing it within a producer, as here we know the table is definitely done across all producer splits.

Reply:
Also make sure you use a scope guard to avoid leaking if a panic happens.